From f3f54c3798a47906d4d44bcf6302e52f3f4cd7b4 Mon Sep 17 00:00:00 2001
From: Abhijith Das <adas@redhat.com>
Date: Fri, 15 Aug 2014 19:56:42 -0400
Subject: [fs] gfs2: replace kmalloc - __vmalloc / memset 0

Message-id: <1408132603-52384-17-git-send-email-adas@redhat.com>
Patchwork-id: 87816
O-Subject: [RHEL7.1 PATCH 16/17] GFS2: replace kmalloc - __vmalloc / memset 0
Bugzilla: 1026406
RH-Acked-by: Robert S Peterson <rpeterso@redhat.com>
RH-Acked-by: Steven Whitehouse <swhiteho@redhat.com>

Use kzalloc and __vmalloc __GFP_ZERO for clean sd_quota_bitmap allocation.

Resolves: rhbz#1026406
Signed-off-by: Fabian Frederick <fabf@skynet.be>
Signed-off-by: Steven Whitehouse <swhiteho@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/gfs2/quota.c b/fs/gfs2/quota.c
index 91bfebb..aedd1d6 100644
--- a/fs/gfs2/quota.c
+++ b/fs/gfs2/quota.c
@@ -1242,14 +1242,13 @@ int gfs2_quota_init(struct gfs2_sbd *sdp)
  bm_size = DIV_ROUND_UP(sdp->sd_quota_slots, 8 * sizeof(unsigned long));
  bm_size *= sizeof(unsigned long);
  error = -ENOMEM;
- sdp->sd_quota_bitmap = kmalloc(bm_size, GFP_NOFS|__GFP_NOWARN);
+ sdp->sd_quota_bitmap = kzalloc(bm_size, GFP_NOFS | __GFP_NOWARN);
  if (sdp->sd_quota_bitmap == NULL)
-  sdp->sd_quota_bitmap = __vmalloc(bm_size, GFP_NOFS, PAGE_KERNEL);
+  sdp->sd_quota_bitmap = __vmalloc(bm_size, GFP_NOFS |
+       __GFP_ZERO, PAGE_KERNEL);
  if (!sdp->sd_quota_bitmap)
   return error;
 
- memset(sdp->sd_quota_bitmap, 0, bm_size);
-
  for (x = 0; x < blocks; x++) {
   struct buffer_head *bh;
   const struct gfs2_quota_change *qc;
-- 
1.7.1